data/transactions/logic/eval_test.go

```diff
@@ -154,9 +164,20 @@ func (ep *EvalParams) reset() {
 	for i := range ep.TxnGroup {
 		ep.TxnGroup[i].ApplyData = transactions.ApplyData{}
 	}
-	ep.created = &resources{}
+	if ep.available != nil {
+		ep.available.apps = nil
+		ep.available.asas = nil
+		// reinitialize boxes because evaluation can add box refs for app creates.
+		available := NewEvalParams(ep.TxnGroup, ep.Proto, ep.Specials).available
+		if available != nil {
+			ep.available.boxes = available.boxes
+		}
+	}
+	ep.readBudgetChecked = false
```

Comment:
@jannotti Checking:  Is `ioBudget` intentionally _not_ reset here?

```suggestion
	ep.readBudgetChecked = false
	ep.ioBudget = 0
```

Reply:
It's fine either way.  The ioBudget is not calculated until the first app is evaluated.  By resetting readBudgetChecked to false, the next call to EvalContract will calculate ioBudget and check it.

Reply:
This thinking, by the way, makes me wonder if we couldn't push a bit more work into the same place, thereby reducing the need to `reset()` in tests at all. (That is, we could intialize almost the entire ep in EvalContract.) But I didnt want want to make that big of a change now.

Reply:
Gotcha and makes sense to avoid enlarging scope - resolving.